### PR TITLE
Fix trace context in event payload

### PR DESCRIPTION
**Pull request by antonpirker**
Make sure that always a `trace` context is added to the event payload. But also make sure that if there is already a `trace` context in the event, do not overwrite it. (This used to be the behavior before tracing without performance. See: https://github.com/getsentry/sentry-python/blob/1.25.1/sentry_sdk/scope.py#L420)

### CODE REVIEW

File: sentry_sdk/scope.py [hunk 1]
```diff
@@ -605,11 +605,14 @@ def _drop(cause, ty):
...
-        else:
-            contexts["trace"] = self.get_trace_context()
+        if contexts.get("trace") is None:
+            if has_tracing_enabled(options):
+                if self._span is not None:
+                    contexts["trace"] = self._span.get_trace_context()
+                else:
+                    contexts["trace"] = self.get_trace_context()
+            else:
+                contexts["trace"] = self.get_trace_context()
```
Comment (sentrivana) [inline sentry_sdk/scope.py:615]: Could simplify this, should be equivalent
```suggestion
            if has_tracing_enabled(options) and self._span is not None:
                contexts["trace"] = self._span.get_trace_context()
            else:
                contexts["trace"] = self.get_trace_context()
```

Comment (sentrivana) [reply sentry_sdk/scope.py:615]: Also wondering if we need the `has_tracing_enabled(options)` check at all? Wouldn't it be enough to check if there's a `self._span`?

**Review by sentrivana — APPROVED** [review]
Looks good, please consider the small suggestion and comment

**Comment by antonpirker** [comment]
We also have spans present if tracing is disabled, so the check needs to be there.

**Comment by antonpirker** [comment]
Thanks for the better if, way nicer to read

**Pull request merged**